cfme/containers/provider/__init__.py

```diff
@@ -355,6 +361,45 @@ def step(self):
         sel.click(InfoBlock('Overview', 'Topology'))
 
 
+class AdHocMetricsView(BaseLoggedInPage):
+    filter_dropdown = SelectorDropdown('uib-tooltip', 'Filter by')
+    apply_btn = form_buttons.apply_filters
+    selected_filter = None
+
+    @property
+    def is_displayed(self):
+        return False
+
+    def wait_for_filter_option_to_load(self):
+        wait_for(lambda: bool(self.filter_dropdown.items), delay=5, num_sec=60)
+
+    def wait_for_results_to_load(self):
+        wait_for(lambda: bool(int(Text(self, 'h5.ng-binding').text.split()[0])),
+                 delay=5, num_sec=60)
+
+    def apply_filter(self):
+        form_buttons._fill_fb_bool(self.apply_btn, True)
+
+    def set_filter(self, desired_filter):
+        self.selected_filter = desired_filter
+        self.filter_dropdown.fill_with(desired_filter)
+
+    def get_random_filter(self):
+        return str(random.choice(self.filter_dropdown.items))
+
+    def get_total_results_count(self):
+        return int(Text(self, 'h5.ng-binding').text.split()[0])
```

Comment:
Same here. Besides I see a DRY opportunity between `wait_for_results_to_load` and `get_total_results_count`.